components/script/dom/window.rs

```diff
@@ -996,7 +996,7 @@ impl Window {
                 page_clip_rect: self.page_clip_rect.get(),
             },
             document: self.Document().upcast::<Node>().to_trusted_node_address(),
-            document_stylesheets: document.stylesheets().clone(),
```

Comment:
This clone here is used to clone a reference, so it's a pretty cheap operation. But your clone in `fn stylesheets()` clones an entire Vec, which is pretty expensive.